lib/rubocop/cop/naming/predicate_name.rb

```diff
@@ -61,7 +61,7 @@ def on_def(node)
         private
 
         def allowed_method_name?(method_name, prefix)
-          !method_name.start_with?(prefix) ||
+          !method_name.match(/^#{prefix}[^0-9]/) ||
```

Comment:
I'm not sure digits were the only characters that we should be mindful of. Perhaps you can check what exactly was allowed there.

Reply:
https://docs.ruby-lang.org/en/trunk/syntax/methods_rdoc.html#label-Method+Names

> Method names may be one of the operators or must start a letter or a character with the eight bit set. It may contain letters, numbers, an _ (underscore or low line) or a character with the eight bit set. 

> Ruby programs must be written in a US-ASCII-compatible character set such as UTF-8, ISO-8859-1 etc. In such character sets if the eight bit is set it indicates an extended character. Ruby allows method names and other identifiers to contain such characters. Ruby programs cannot contain some characters like ASCII NUL (\x00).

The description isn't very precise to what they mean by a "letter".  But seeing that the eight bit is used to denote an extended character set I think that means within the US-ASCII range; `a-z` and `A-Z`.

So the recipe for valid starting character is `a-z`, `A-Z`, any character with the eight bit set. Also one would have to consider any of the valid operators. At least that is how I interpret the description from the documentation.

Reply:
> Identifiers consist of alphabets, decimal digits, and the underscore character, and begin with a alphabets(including underscore).

> Method_name should be either identifier or re-definable operators (e.g. ==, +, -, etc.).

Ref: https://ruby-doc.org/docs/ruby-doc-bundle/Manual/man-1.4/syntax.html

Which means after removing the prefix `is_` from `is_<predicate>?` the remaining `<predicate>` should start with either `a-z`, `A-Z` or `_`. Or in other words should not start with `0-9`.

> Perhaps you can check what exactly was allowed there.

The cop removed prefix and showed the rest as suggested method-name.